tiledb/api/src/non_empty_domain.rs

```diff
@@ -0,0 +1,111 @@
+use std::fmt::{Debug, Formatter, Result as FmtResult};
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use crate::datatype::Datatype;
+
+#[derive(Serialize, Deserialize)]
+pub enum Range {
+    UInt8Range(u8, u8),
+    UInt16Range(u16, u16),
+    UInt32Range(u32, u32),
+    UInt64Range(u64, u64),
+    Int8Range(i8, i8),
+    Int16Range(i16, i16),
+    Int32Range(i32, i32),
+    Int64Range(i64, i64),
+    Float32Range(f32, f32),
+    Float64Range(f64, f64),
+    VarSizeRange(Box<[u8]>, Box<[u8]>),
```

Comment:
Is this meant to be attached to a specific datatype?
From what I've come to understand the var size unit could be any of the primitive types, right?

Reply:
Correct. I was just being lazy. I'll go back and add `Var${Type}Range` and do it properly now that I have the basics done.

Reply:
Fixed.